pkg/gce-cloud-provider/compute/fake-gce.go

```diff
@@ -212,11 +212,6 @@ func (cloud *FakeCloudProvider) ValidateExistingDisk(ctx context.Context, resp *
 			params.DiskType, respType[len(respType)-1])
 	}
 
-	// We are assuming here that a multiWriter disk could be used as non-multiWriter
-	if multiWriter && !resp.GetMultiWriter() {
```

Comment:
I think we need to keep this check, to validate that we aren't trying to use a non-multiwriter disk when we need a multiwriter one?